tools/probes/probes_main.c

```diff
@@ -0,0 +1,288 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
+
+/*
+ * Probes will extract data for several probe points in one stream
+ * with extra headers. This app will read the resulting file,
+ * strip the headers and create wave files for each extracted buffer.
+ *
+ * Usage to parse data and create wave files: ./sof-probes -p data.bin
+ *
+ */
+
+#include <ipc/probe.h>
+#include <sof/math/numbers.h>
+#include "wave.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#define APP_NAME "sof-probes"
+
+#define DATA_READ_LIMIT 1000
+#define FILES_LIMIT	8
+
+struct wave_files {
+	FILE *fd;
+	uint32_t buffer_id;
+	uint32_t size;
+	struct wave header;
+};
+
+enum p_state {
+	READY = 0,
+	SYNC,
+	CHECK
+};
+
+uint32_t sample_rate[15] = {
```

Comment:
Why no `static`? It is used only in this file. Also I guess `15` is unnecessary, compiler should fill it.

Reply:
Fixed